app/controllers/admin_controller.rb

```diff
@@ -0,0 +1,45 @@
+class AdminController < ApplicationController
+  def summary
+    if Order.all.any?
+      @popular_movies = ordered_showtimes.map(&:movie).uniq[0..2]
+      @popular_times = ordered_times.uniq[0..2]
+      @total_revenue = Order.all.map(&:total_cost).sum
+      @daily_total_revenues = days_of_the_week.map { |day| day_with_revenue(day) }
+    else
+      @no_orders = true
+    end
+  end
+
+  def orders
+    @orders = Order.all
+  end
```

Comment:
Wasn't sure if this was the ideal pattern or if I should instead have summary and orders controllers with a show and index route respectively.